src/heading.js

```diff
@@ -27,46 +25,120 @@ export default class Heading extends Plugin {
 	 * @inheritDoc
 	 */
 	static get requires() {
-		return [ HeadingEngine ];
+		return [ Paragraph, HeadingEngine ];
 	}
 
 	/**
 	 * @inheritDoc
 	 */
 	init() {
 		const editor = this.editor;
-		const command = editor.commands.get( 'heading' );
-		const options = command.options;
-		const collection = new Collection();
-
-		// Add options to collection.
-		for ( const { id, label } of options ) {
-			collection.add( new Model( {
-				id, label
+		const dropdownItems = new Collection();
+		const options = this._getLocalizedOptions();
+		const commands = [];
+		let defaultOption;
+
+		for ( let option of options ) {
+			// Add the option to the collection.
+			dropdownItems.add( new Model( {
+				modelElement: option.modelElement,
+				label: option.title
```

Comment:
This `label = title` doesn't look good.

I think that the name of the option property should be as generic as possible because we'll have the same problem all around our code. I wonder what's better – label, title or perhaps name. We have the same problem in `ImageStyles` and it would be good to find a naming convention fitting all such scenarios.